src/commands/tileindex-validate/tileindex.validate.ts

```diff
@@ -381,23 +422,20 @@ export async function extractTiffLocations(
           return null;
         }
 
-        const centerX = (bbox[0] + bbox[2]) / 2;
-        const centerY = (bbox[1] + bbox[3]) / 2;
-        // bbox is not epsg:2193
         const targetProjection = Projection.get(2193);
         const sourceProjection = Projection.get(sourceEpsg);
 
-        const [x, y] = targetProjection.fromWgs84(sourceProjection.toWgs84([centerX, centerY]));
-        if (x == null || y == null) {
+        const [ulX, lrY, lrX, ulY] = reprojectIfNeeded(bbox, sourceProjection, targetProjection);
+
+        if (ulX == null || ulY == null || lrX == null || lrY == null) {
           logger.error(
             { reason: 'Failed to reproject point', source: tiff.source },
             'Reprojection:ExtracTiffLocations:Failed',
           );
           return null;
         }
 
-        // Tilename from center
-        const tileName = getTileName(x, y, gridSize);
+        const covering = [...iterateMapSheets([ulX, ulY, lrX, lrY], gridSize)] as string[];
```

Comment:
would be good to just return a full `string[]` rather than iterating.

Reply:
out of curiosity, what was the intent behind implementing this as an iterator at the start?

Reply:
I believe I copy and pasted this function from somewhere it was doing a async process as it iterated, it may have also broken early rather than collecting the full list, but my memory is a little hazy as to exactly where it came from.

Reply:
done